runner/app/live/pipelines/streamdiffusion.py

```diff
@@ -74,25 +79,121 @@ async def get_processed_video_frame(self) -> VideoOutput:
+                if new_params.width != self.params.width or new_params.height != self.params.height:
+                    await self._change_resolution_locked(new_params)
+                    return
+
+            logging.info(f"Resetting diffuser for params change")
+
+            self.pipe = await asyncio.to_thread(load_streamdiffusion_sync, new_params)
+            self.params = new_params
+            self.first_frame = True
+
+    async def _change_resolution_locked(self, new_params: StreamDiffusionParams):
+        """Internal method to change resolution. Must be called while holding the pipeline lock."""
+        logging.info(f"Changing resolution from {self.params.width}x{self.params.height} to {new_params.width}x{new_params.height}")
+        
+        # Stop the current pipeline with timeout protection
+        try:
+            await asyncio.wait_for(self._stop_locked(), timeout=8.0)
+        except asyncio.TimeoutError:
+            logging.warning("Timeout during resolution change stop, forcing cleanup")
+            self.pipe = None
+            if torch.cuda.is_available():
+                try:
+                    torch.cuda.empty_cache()
+                except Exception as e:
+                    logging.warning(f"Error during forced CUDA cleanup: {e}")
+        
+        # Initialize with new parameters
         self.pipe = await asyncio.to_thread(load_streamdiffusion_sync, new_params)
         self.params = new_params
         self.first_frame = True
+        
+        logging.info("Resolution change complete")
 
     async def stop(self):
+        """Stop the pipeline. Public method that acquires the lock."""
+        try:
+            # Add a timeout to prevent hanging indefinitely
+            await asyncio.wait_for(self._stop_with_lock(), timeout=10.0)
+        except asyncio.TimeoutError:
+            logging.error("Timeout occurred while stopping StreamDiffusion pipeline")
+            # Force cleanup even if timeout occurred
+            self.pipe = None
+            if torch.cuda.is_available():
+                try:
+                    torch.cuda.empty_cache()
+                except Exception as e:
+                    logging.warning(f"Error during CUDA cache cleanup: {e}")
+        except Exception as e:
+            logging.error(f"Error stopping StreamDiffusion pipeline: {e}")
+            self.pipe = None
+
+    async def _stop_with_lock(self):
+        """Helper method to acquire lock and stop."""
+        async with self._pipeline_lock:
+            await self._stop_locked()
+
+    async def _stop_locked(self):
+        """Internal method to stop the pipeline. Must be called while holding the pipeline lock."""
         logging.info("Stopping StreamDiffusion pipeline")
-        self.pipe = None
+        
+        # Clear the frame queue with timeout protection
+        try:
+            queue_clear_count = 0
+            while not self.frame_queue.empty() and queue_clear_count < 100:  # Limit iterations
```

Comment:
Let's try to get rid of this complex clean-up logic as I think `streamdiffusion` will be able to reload just fine. I'll test it on my dev box and update this if it works.